hibernate-core/src/main/java/org/hibernate/engine/spi/ActionQueue.java

```diff
@@ -1302,11 +1297,9 @@ private void addParentChildEntityNameByPropertyAndValue(AbstractEntityInsertActi
 					if ( !batchIdentifier.getEntityName().equals( entityName ) ) {
 						batchIdentifier.getParentEntityNames().add( entityName );
 					}
-					if ( value != null ) {
-						String valueClass = value.getClass().getName();
-						if ( !valueClass.equals( entityName ) ) {
-							batchIdentifier.getParentEntityNames().add( valueClass );
-						}
+					String valueClass = value.getClass().getName();
+					if ( !valueClass.equals( entityName ) ) {
+						batchIdentifier.getParentEntityNames().add( valueClass );
```

Comment:
The above is leftover of https://github.com/hibernate/hibernate-orm/pull/3581. We don't need nullness checking here for the only upstream method ensures that the value is never null.

Reply:
As I mentioned in the previous PR, that's wrong and there's failing tests if you make this change.

Reply:
yeah, I realized for I found the broken TumTest case. Reverted back.